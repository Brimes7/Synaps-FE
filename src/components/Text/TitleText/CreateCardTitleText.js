import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

/**
 * Title Text
 *
 * @component
 * @example
 *  <TitleText text={"Create Deck"} />
 *
 */
export const CreateCardTitleText = ({text}) => {
  return <StyledTitle>{text}</StyledTitle>;
};

const StyledTitle = styled.div`
  color: #2a685b;
  font-family: Source Sans Pro;
  font-style: normal;
  font-weight: 900;
  font-size: 45px;
  line-height: 24px;
  font-weight: bold;
  text-align: left;
  margin-bottom: 30px;
  margin-top: 30px;
`;

CreateCardTitleText.propTypes = {
  text: PropTypes.string.isRequired,
};
