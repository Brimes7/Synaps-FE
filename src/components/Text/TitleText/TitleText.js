import React from 'react';
import PropTypes from 'prop-types';
import styled, { keyframes } from 'styled-components';

/**
 * Title Text
 *
 * @component
 * @example
 *  <TitleText text={"Create Deck"} />
 *
 * @param text
 * @param color
 * @return {*}
 */
export const TitleText = ( { text, color = '#2A685B' } ) => {
  return (
    <StyledTitleContainer>
      <StyledTitle color={ color }>{ text }</StyledTitle>
    </StyledTitleContainer>
  );
};

const StyledTitleContainer = styled.div`
width: 100%;
`;

const StyledTitle = styled.div`
  font-family: Source Sans Pro;
  color: ${ props => props.color };
  font-weight: 900;
  font-size: 45px;
  margin-left: 10%;
  margin-top: 24px;
  text-align: left;
  margin-bottom: 36px;
`;

TitleText.propTypes = {
  text: PropTypes.string.isRequired,
  color: PropTypes.string,
};

/**
 *
 * @type {Keyframes}
 */
const animation = keyframes`
  0% {
  opacity: 0;
  }
100%{
  opacity: 1;
}
`;