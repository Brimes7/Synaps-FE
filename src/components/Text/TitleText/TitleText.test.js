import React from 'react';
import renderer from 'react-test-renderer';

import {TitleText} from './TitleText.js';



test('StyledTitleText render correctly', () => {
    const tree = renderer.create(<TitleText />).toJSON();
    expect(tree).toMatchSnapshot();
});