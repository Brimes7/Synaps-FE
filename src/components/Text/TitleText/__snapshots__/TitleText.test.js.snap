// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledTitleText render correctly 1`] = `
.c0 {
  color: #888888;
  font-weight: bold;
  font-size: 45px;
  margin-left: 10%;
  margin-top: 24px;
  text-align: left;
  margin-botom: 36px;
}

<div
  className="c0"
/>
`;
